src/components/Session/withAuthentication.js

```diff
@@ -1,42 +1,31 @@
-import React from 'react';
+import React, {useState, useEffect} from 'react';
 
 import AuthUserContext from './context';
 import { withFirebase } from '../Firebase';
 
 const withAuthentication = Component => {
```

Comment:
this is cool. I recently opened an issue with a question regarding this and I think your implementation here has given me a good idea of what to do!